### PR TITLE
Add generic resource sharing test utilizing test account cluster fixture.

### PR DESCRIPTION
**Pull request by rohinb2**
I introduce a sharing test for sharing resources generically with another user. This uses another local cluster that is logged in as a test user.

### CODE REVIEW

**Comment by rohinb2** [comment]
Current dependencies on/for this PR:
* main
  * **PR #154** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/154?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #155** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/155?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #156** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/156?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/156?utm_source=stack-comment).

File: tests/test_resources/test_clusters/conftest.py [hunk 1]
```diff
@@ -411,24 +411,30 @@ def local_docker_cluster_telemetry_public_key(request, detached=True):
...
+        container_name = "rh-slim-test-acct"
+        dir_name = "public-key-auth"
+        detached = request.config.getoption("--detached")
+        den_auth = "den_auth" in request.keywords
+        keypath = str(
+            Path(
+                rh.configs.get("default_keypair", DEFAULT_KEYPAIR_KEYPATH)
+            ).expanduser()
+        )
+        local_ssh_port = BASE_LOCAL_SSH_PORT + 4
```
Comment (DenisYay) [inline tests/test_resources/test_clusters/conftest.py:431]: At some point we'll have to do something about the hardcoded offsets in the fixtures..

File: tests/test_resources/test_resource.py [hunk 2]
```diff
@@ -80,6 +88,36 @@ def test_save_and_load(self, resource):
...
+            from runhouse.globals import rns_client
+
+            resource.rns_address = rns_client.local_to_remote_address(
+                resource.rns_address
+            )
+
+        # Test saving, then share with test user
+        resource.save()
+
+        resource.share(
```
Comment (DenisYay) [inline tests/test_resources/test_resource.py:109]: Do we need to assert it is not shared before and indeed shared after?

**Comment by rohinb2** [comment]
### Merge activity

* **Nov 20, 4:26 PM**: @rohinb2 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/156).

File: tests/test_resources/test_clusters/conftest.py (hunk 1 above)
Comment (dongreenberg) [reply tests/test_resources/test_clusters/conftest.py:431]: I think docker network is the durable answer

**Pull request merged**